common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java

```diff
@@ -59,8 +60,10 @@ public class TransportResponseHandler extends MessageHandler<ResponseMessage> {
   /** Records the time (in system nanoseconds) that the last fetch or RPC request was sent. */
   private final AtomicLong timeOfLastRequestNs;
 
-  private final ScheduledExecutorService pushTimeoutChecker;
   private final long pushTimeoutCheckerInterval;
+  private static final ScheduledExecutorService pushTimeoutChecker =
+      ThreadUtils.newDaemonThreadPoolScheduledExecutor("push-timeout-checker-", 16);
```

Comment:
Better to make numThreads configurable. We can sync and initialize pushTimeoutChecker in the constructor

Reply:
Done